src/topp/FidoAdapter.cpp

```diff
@@ -118,6 +121,58 @@ public TOPPBase
   typedef boost::bimap<String, String> StringBimap;
   
   StringBimap sanitized_accessions_; // protein accessions
+
+  /**************** Only used if greedy_group_resolution is set ***************/
+  // build bipartite graph as two maps (adjacency "lists"):
+  // ProtGroups-Indices <-> PepID-Indices
+  // so we get bidirectional connectivity
+  // We always take first PepHit from PepID, because those were used for
+  // inference in Fido
+  typedef map<Size, set<Size> > IndexMap;
+  
+  IndexMap indist_prot_grp_to_pep_;
+  IndexMap pep_to_indist_prot_grp_;
+  
+  // represents the middle layer of a tripartite graph
+  // consists of single protein accessions and their mapping to the groups
+  // indices
+  map<String, Size> prot_acc_to_indist_prot_grp_;
+  
+  // represents a connected component of the bipartite graph
+  struct ConnCompStruct
```

Comment:
why not name it "ConnectedComponent"?
(I think it's generally bad style to repeat what a name refers to in the name, e.g. "struct ...Struct", "registerFlag_(...Flag, ...)".)